hyperspy/model.py

```diff
@@ -891,6 +899,312 @@ def _model_function(self, param):
         to_return = self.__call__(non_convolved=False, onlyactive=True)
         return to_return
 
+    @property
+    def nonlinear_parameters(self):
+        "List all active nonlinear parameters"
+        nonlinear = []
+        for comp in self:
+            if comp.active:
+                for para in comp.parameters:
+                    if not para._is_linear:
+                        nonlinear.append(para)
+        return nonlinear
+
+    @property
+    def linear_parameters(self):
```

Comment:
Yes, we should rename it. No, it probably doesn't need to be public, although it can be convenient to check which parameters are linear. Maybe we could instead implement a display of linear parameters into `print_current_values` in the future.